pinecone/client.go

```diff
@@ -754,53 +763,99 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
 	return nil
 }
 
+// ConfigureIndexParams contains parameters for configuring an index. For both pod-based
+// and serverless indexes you can configure the DeletionProtection status for an index.
+// For pod-based indexes you can also configure the number of Replicas and the PodType.
+// Each of the fields are optional, but at least one field must be set.
+// See [scale a pods-based index] for more information.
+//
+// Fields:
+//   - PodType: (Optional) The pod size to scale the index to. For a "p1" pod type,
+//     you could pass "p1.x2" to scale your index to the "x2" size, or you could pass "p1.x4"
+//     to scale your index to the "x4" size, and so forth. Only applies to pod-based indexes.
+//   - Replicas: (Optional) The number of replicas to scale the index to. This is capped by
+//     the maximum number of replicas allowed in your Pinecone project. To configure this number,
+//     go to [app.pinecone.io], select your project, and configure the maximum number of pods.
+//   - DeletionProtection: (Optional) // DeletionProtection determines whether [deletion protection]
+//     is "enabled" or "disabled" for the index. When "enabled", the index cannot be deleted. Defaults to "disabled".
+//
+// Example:
+//
+//	    ctx := context.Background()
+//
+//	    clientParams := pinecone.NewClientParams{
+//		       ApiKey:    "YOUR_API_KEY",
+//		       SourceTag: "your_source_identifier", // optional
+//	    }
+//
+//	    pc, err := pinecone.NewClient(clientParams)
+//	    if err != nil {
+//	        log.Fatalf("Failed to create Client: %v", err)
+//	    } else {
+//		       fmt.Println("Successfully created a new Client object!")
+//	    }
+//
+//	    idx, err := pc.ConfigureIndex(ctx, "my-index", &ConfigureIndexParams{ DeletionProtection: "enabled", Replicas: 4 })
+//
+// [app.pinecone.io]: https://app.pinecone.io
+// [scale a pods-based index]: https://docs.pinecone.io/guides/indexes/configure-pod-based-indexes
+type ConfigureIndexParams struct {
+	PodType            string
+	Replicas           int32
+	DeletionProtection control.DeletionProtection
+}
+
 // ConfigureIndex is used to [scale a pods-based index] up or down by changing the size of the pods or the number of
-// replicas.
+// replicas, or to enable and disable deletion protection for an index.
```

Comment:
Sorry for being behind the times, but so now you can call `ConfigureIndex` on a serverless index?

Reply:
(Should we add a test that tries to call `configureIndex` on a serverless index, but passes in `replicas` and `pods` (which obvi aren't things on serverless indexes), so that the test should expect a failure? I think we should!

Reply:
Yes, let me add a couple tests.